drivers: i3c: rename dcr i2c mode macro to lvr

XenuIsWatching opened the pull request:
This renames the I2C 'DCR' mode to 'LVR' as that is the variable it
should be looking at and not the dcr value. This also fixes the get
'lvr' mode argument.

XenuIsWatching committed:
drivers: i3c: rename dcr i2c mode macro to lvr

This renames the I2C 'DCR' mode to 'LVR' as that is the variable it
should be looking at and not the dcr value. This also fixes the get
'lvr' mode argument.

Signed-off-by: Ryan McClelland <ryanmcclelland@meta.com>

dkalowsk reviewed (changes requested):
Nothing wrong with the change as is, just an additional request.  This is going to break an API for folks on earlier revisions, so... requesting that you add a note in the release notes now as this comes in.

XenuIsWatching committed:
doc: releases: add i3c dcr/lvr rename note

Add a note to the v3.6 release notes about renaming the I3C_DCR_I2C_*
to I3C_LVR_I2C_*.

Signed-off-by: Ryan McClelland <ryanmcclelland@meta.com>

XenuIsWatching commented:
> Nothing wrong with the change as is, just an additional request. This is going to break an API for folks on earlier revisions, so... requesting that you add a note in the release notes now as this comes in.

done

dkalowsk reviewed (dismissed):
Thanks!

henrikbrixandersen commented on doc/releases/release-notes-3.6.rst:
```diff
@@ -145,6 +145,9 @@ Drivers and Sensors
 
 * I3C
 
+  * The Legacy Virtual Register defines have been renamed from I3C_DCR_I2C_* to
+    I3C_LVR_I2C_*.
```
You need to enclose the names in double backticks.

XenuIsWatching replied:
done
